riscv/devices.cc

```diff
@@ -12,6 +12,22 @@ void bus_t::add_device(reg_t addr, abstract_device_t* dev)
   devices[addr] = dev;
 }
 
+void bus_t::set_interrupt_controller(abstract_interrupt_controller_t* intdev)
```

Comment:
Why is a bus associated with an interrupt controller?  Is this just a software-engineering simplification so that all devices on a bus are associated with the same interrupt controller?

Reply:
It's just a software-engineering simplification.

I guess a more flexible way would be to have interrupt controller associated with a device. This will allow multiple interrupt controllers on the same bus where different sets of devices are associated with interrupt controller.